<launch>
    <group ns="nightray">
        <!-- aruco ros params -->
        <arg name="markerId" default="5" />
        <arg name="markerSize" default="0.15" /> <!-- in meters -->
        <arg name="marker_frame" default="marker_frame" />
        <arg name="camera_frame" default="oak_d_pro_w_rgb_camera_frame" />
        <arg name="ref_frame" default="" /> <!-- leave empty and the pose will be published wrt param parent_name -->
        <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
        
        <!-- controller params -->
        <arg name="x_p_gain" default="0.8" />
        <arg name="y_p_gain" default="0.8" />
        <!-- <arg name="x_p_gain" default="0.0" />
        <arg name="y_p_gain" default="0.0" /> -->
        <arg name="x_pos_goal" default="0.9" />
        <arg name="yaw_p_gain" default="2.0" />
        <!-- <arg name="yaw_vy_p_gain" default="0.25" /> -->
        <!-- <arg name="yaw_p_gain" default="0.0" /> -->
        <arg name="yaw_vy_p_gain" default="1.0" />

        <arg name="max_v" default="0.125" />
        <arg name="max_yaw_rate" default="0.25" />

        <arg name="takeoff_height" default="1.0" />

        <arg name="drone_frame" default="" /> <!-- blank to use camera frame -->

        <node pkg="aruco_detect" type="aruco_detect" name="aruco_single">
            <remap from="camera_info" to="camera/rgb/camera_info" />
            <remap from="camera" to="camera/rgb/image" />
            <param name="dictionary" value="0" />
            <param name="publish_images" value="true" />
            <param name="fiducial_len" value="$(arg markerSize)" />
            <!-- <param name="reference_frame" value="$(arg ref_frame)" />  -->
            <!-- frame in which the marker pose will be refered -->
            <!-- <param name="camera_frame" value="$(arg camera_frame)" />
            <param name="marker_frame" value="$(arg marker_frame)" /> -->
            <!-- <param name="corner_refinement" value="$(arg corner_refinement)" /> -->
        </node>
        
        <node pkg="draconis_aruco" type="aruco_pcontrol.py" name="aruco_pcontrol" output="screen">
            <param name="x_p_gain" value="$(arg x_p_gain)" />
            <param name="x_pos_goal" value="$(arg x_pos_goal)" />
            <param name="y_p_gain" value="$(arg y_p_gain)" />
            <param name="yaw_p_gain" value="$(arg yaw_p_gain)" />
            <param name="drone_frame" value="$(arg drone_frame)" />
            <param name="yaw_vy_p_gain" value="$(arg yaw_vy_p_gain)" />
            <param name="max_v" value="$(arg max_v)" />
            <param name="max_yaw_rate" value="$(arg max_yaw_rate)" />
            <param name="takeoff_height" value="$(arg takeoff_height)" />
            <remap from="/target" to="/mavros/setpoint_raw/local" />
        </node>
    </group>
</launch>
